lua/rocks_treesitter/highlight.lua

```diff
@@ -36,11 +36,17 @@ local function is_installed(lang)
 end
 
 ---@param lang string
----@return Rock[]
-local function find_parser_rock(lang)
+---@param callback fun(rock: Rock[])
+local function find_parser_rock(lang, callback)
     local rock_name = "tree-sitter-" .. lang
     local rocks = api.try_get_cached_rocks()
-    return rocks[rock_name] or {}
+    if rocks and rocks[rock_name] then
+        callback(rocks[rock_name])
```

Comment:
Should you not return after this call? If not, the callback will be called twice from the cache (if it is populated) and luarocks?

Reply:
Good call. Pushed a fix on `main`.